packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -3,40 +3,54 @@ import React from 'react';
 // This variable will be true once the server-side hydration is completed.
 let hydrationCompleted = false;
 
+function deepEqual(a, b) {
+  return a.length === b.length && a.every((item, index) => item === b[index]);
+}
+
 function useMediaQuery(queryInput, options = {}) {
-  const query = queryInput.replace('@media ', '');
+  const multiple = Array.isArray(queryInput);
```

Comment:
Just seeing this since it was not obvious that any implementation changed from the PR title.

What is the use case for this? media queries already support combining multiple queries with boolean logic. This bloats the code a lot. It's nowhere documented that this doesn't support changing the number of queries provided.

Reply:
> What is the use case for this?

The use case is useWidth. I have first tried to implement it with the existing `useMediaQuery.js` logic. It wasn't possible. It was making the component render x2 x card(breakpoints), so around 10 renders. 

> media queries already support combining multiple queries with boolean logic.

And yet, it returns a single boolean.

>  This bloats the code a lot.

I have seen worse 
![Capture d’écran 2019-06-08 à 15 24 29](https://user-images.githubusercontent.com/3165635/59148489-8efc2380-8a01-11e9-8cc6-de03cda5d853.png)

> doesn't support changing the number of queries provided.

It should be supported.

Reply:
>  It was making the component render x2 x card(breakpoints), so around 10 renders.

Did you profile it?

> And yet, it returns a single boolean

Can always use multiple hooks. No need to overload the code.

> > doesn't support changing the number of queries provided.

> It should be supported.

It is passed as a dependency errors to hooks.

Reply:
> Did you profile it?

The withWith() higher-order component was making his wrapped component render 10 times. The performance impact is not the only dimension to consider. The more renders, the harder debugging experience for people using this higher order component.

Reply:
This change was released yet. It's not too late to revert it if you don't feel confident with the direction. You can play with the alternative here: https://material-ui.com/components/use-media-query/#migrating-from-withwidth. I doubt we can better than this new strategy, but who knows? Maybe if we ignore the dynamic breakpoint occurrence, and that we use an intermediary pure component to remove the redundant renders, it would work nicely.

Reply:
An interesting implementation to benchmark against: https://github.com/streamich/use-media/blob/master/src/index.ts.

Reply:
Ok, I will give it a second try.

Reply:
I can't find a way to deduplicate the renders.

Notice that I haven't updated the `useMediaQuery` TypeScript signature nor documented the array existance. What do you of this middle ground solution? We revert the useWidth migration example so we can, later, continue iterate on this topic without being lock-in with the array behind a public API, used by people.